### PR TITLE
docs/term: add information to make it work with WSL

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Only problem is that it breaks Tab and Backspace for me.

### CODE REVIEW

File: docs/config/lua/config/term.md [hunk 1]
```diff
@@ -22,3 +22,5 @@ as colored underlines, styled underlines (eg: undercurl).  If the system
 you are using has a relatively outdated ncurses installation, the `wezterm`
 terminfo will also enable italics and true color support.
 
+If you are using WSL in wezterm you need to set the `WSLENV` variable in Windows to `TERM` otherwise WSL will default to `xterm-256color`.
+See [this Microsoft blog post](https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/#what-are-environment-variables) for more information on how to do that.
```
Comment (SuperSandro2000) [inline docs/config/lua/config/term.md:26]:
````suggestion
See [this Microsoft blog post](https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/#what-are-environment-variables) for more information on how `WSLENV` works.

Add the following snippet to your config to make it work if you manually installed the TERM definition.

```lua
  set_environment_variables={
    WSLENV='TERM'
  },
  term = "wezterm",
```

If your package manager installed the TERM definition in a none standard location which will likely be the case if your are using nixpkgs/home-manager/NixOS then you need to set `TERMINFO_DIRS` before your profile is loaded.
The following snippet works if you installed `wezterm.terminfo` with nix into your user profile. Update the path to `TERMINFO_DIRS` to your location.

```lua
  set_environment_variables={
    TERMINFO_DIRS='/home/user/.nix-profile/share/terminfo',
    WSLENV='TERM:TERMINFO_DIRS'
  },
  term = "wezterm",
```
````

Comment (dankamongmen) [reply docs/config/lua/config/term.md:26]: s/none standard/non-standard/g

**Comment by wez** [comment]
Nightly wezterm now automatically adds `TERM`, `COLORTERM`, `TERM_PROGRAM` and `TERM_PROGRAM_VERSION` to `WSLENV` if we're on Windows, or if we're on Linux and we look like we're running under WSL.

Hopefully that means that you can simplify this config down to just your TERMINFO bits!

File: docs/config/lua/config/term.md [hunk 2]
```diff
@@ -22,3 +22,5 @@ as colored underlines, styled underlines (eg: undercurl).  If the system
 you are using has a relatively outdated ncurses installation, the `wezterm`
 terminfo will also enable italics and true color support.
 
+If you are using WSL in wezterm you need to set the `WSLENV` variable in Windows to `TERM` otherwise WSL will default to `xterm-256color`.
```
Comment (SuperSandro2000) [inline docs/config/lua/config/term.md:25]:
```suggestion
If you are using WSL, wezterm will automatically populate `WSLENV` to properly set TERM, COLORTERM, TERM_PROGRAM and TERM_PROGRAM_VERSION.
```

File: docs/config/lua/config/term.md [hunk 3]
```diff
@@ -22,3 +22,16 @@ as colored underlines, styled underlines (eg: undercurl).  If the system
 you are using has a relatively outdated ncurses installation, the `wezterm`
 terminfo will also enable italics and true color support.
 
+If you are using WSL, wezterm will automatically populate `WSLENV` to properly set TERM, COLORTERM, TERM_PROGRAM and TERM_PROGRAM_VERSION.
+See [this Microsoft blog post](https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/#what-are-environment-variables) for more information on how `WSLENV` works.
+
+If your package manager installed the TERM definition in a non-standard location which will likely be the case if your are using nixpkgs/home-manager/NixOS then you need to set `TERMINFO_DIRS` before your profile is loaded.
+The following snippet works if you installed `wezterm.terminfo` with nix into your user profile. Update the path to `TERMINFO_DIRS` to your location.
```
Comment (wez) [inline docs/config/lua/config/term.md:29]:
```suggestion
The following snippet works if you installed `wezterm.terminfo` with nix into your user profile. Update the path to `TERMINFO_DIRS` to match the location on your system.
```

File: docs/config/lua/config/term.md [hunk 4]
```diff
@@ -22,3 +22,16 @@ as colored underlines, styled underlines (eg: undercurl).  If the system
 you are using has a relatively outdated ncurses installation, the `wezterm`
 terminfo will also enable italics and true color support.
 
+If you are using WSL, wezterm will automatically populate `WSLENV` to properly set TERM, COLORTERM, TERM_PROGRAM and TERM_PROGRAM_VERSION.
```
Comment (wez) [inline docs/config/lua/config/term.md:25]:
```suggestion
If you are using WSL, wezterm will automatically populate `WSLENV` to properly set TERM, COLORTERM, TERM_PROGRAM and TERM_PROGRAM_VERSION in the environment when crossing between win32 and WSL environments.
```

File: docs/config/lua/config/term.md [hunk 5]
```diff
@@ -22,3 +22,16 @@ as colored underlines, styled underlines (eg: undercurl).  If the system
...
+See [this Microsoft blog post](https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/#what-are-environment-variables) for more information on how `WSLENV` works.
+
+If your package manager installed the TERM definition in a non-standard location which will likely be the case if your are using nixpkgs/home-manager/NixOS then you need to set `TERMINFO_DIRS` before your profile is loaded.
```
Comment (wez) [inline docs/config/lua/config/term.md:28]:
```suggestion
If your package manager installed the terminfo data in a non-standard location, which will likely be the case if your are using nixpkgs/home-manager/NixOS, then you need to set `TERMINFO_DIRS` in the environment in order for applications to find it.
```

**Review by wez — APPROVED** [review]

**Comment by wez** [comment]
Thanks!

**Pull request merged**